### PR TITLE
SB-1253 ArtifactGroup

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

File: strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.18__ArtifactGroup.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <changeSet id="v1.0.0.18" author="przemyslaw.fusik@gmail.com">
+
+        <o:createClass name="ArtifactGroup" abstract="true">
+            <o:property name="name" type="string" notNull="true"/>
+            <o:property name="artifactEntries" type="linkset" linkedType="ArtifactEntry"/>
+        </o:createClass>
+
+        <o:createClass name="ArtifactIdGroup" extends="ArtifactGroup"/>
+
+        <o:createIndex name="idx_artifact_id_group" on="ArtifactIdGroup" property="name" type="unique" />
```
Comment (sbespalov) [inline strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.18__ArtifactGroup.xml]: @fuss86 have you bean able to check that this constraint triggers only for `ArtifactIdGroup` subclass, and not for all `ArtifactGroup`?

Comment (fuss86) [reply strongbox-db-liquibase/src/main/resources/db/changelog/v1.0.0/v1.0.0.18__ArtifactGroup.xml]: this idx does not exist anymore

File: Jenkinsfile [hunk 2]
```diff
@@ -68,18 +76,9 @@ pipeline {
...
-                                VERSION_ID = "${BRANCH_NAME}-${env.VERSION}"
-                            }
-
-                            sh "mvn versions:set -DnewVersion=${VERSION_ID}"
-
                             sh "mvn deploy" +
                                " -DskipTests" +
-                               " -DaltDeploymentRepository=${SERVER_ID}::default::${SERVER_URL}"
-                        }
+                               " -DaltDeploymentRepository=${SERVER_ID}::default::${SERVER_URL}"                        }
```
Comment (steve-todorov) [inline Jenkinsfile]: Cosmetic: Formatting at the end of the line :)

File: strongbox-db-import/pom.xml [hunk 3]
```diff
@@ -98,32 +78,35 @@
...
                         </goals>
+                        <configuration>
+                            <descriptors>
+                                <descriptor>src/main/assembly/db-schema.xml</descriptor>
+                            </descriptors>
+                        </configuration>                        
                     </execution>
                 </executions>
-            </plugin>
+            </plugin>
```
Comment (carlspring) [inline strongbox-db-import/pom.xml:114]: Cosmetic: Please, remove the extra spaces at the end of line.

File: strongbox-db-import/src/main/java/org/carlspring/strongbox/db/orient/OrientDbConfiguration.java [hunk 4]
```diff
@@ -1,53 +1,77 @@
...
  */
-@Slf4j
 @Configuration
 class OrientDbConfiguration
 {
+    
+    private static final Logger logger = LoggerFactory.getLogger(OrientDbConfiguration.class);
+    
+    @Bean
+    OrientDbServer orientDbServer(OrientDbServerConfiguration serverProperties, OrientDbStudioConfiguration studioProperties) {
```
Comment (carlspring) [inline strongbox-db-import/src/main/java/org/carlspring/strongbox/db/orient/OrientDbConfiguration.java:40]: Cosmetic: Braces on next line, please! (Here and below and in various other places across the pull request).

File: strongbox-db-import/src/main/java/org/carlspring/strongbox/db/orient/OrientDbConfiguration.java [hunk 5]
```diff
@@ -1,53 +1,77 @@
...
+        return new EmbeddedOrientDbServer(studioProperties, serverProperties);
+    }
 
-    @Autowired
-    private OrientDbProperties orientDbProperties;
+    @Bean
+    @ConfigurationProperties(prefix = "strongbox.orientdb.studio")
+    OrientDbStudioConfiguration orientDbStudioProperties() {
```
Comment (carlspring) [inline strongbox-db-import/src/main/java/org/carlspring/strongbox/db/orient/OrientDbConfiguration.java:46]: As above.

File: strongbox-db-import/src/main/java/org/carlspring/strongbox/db/orient/OrientDbConfiguration.java [hunk 6]
```diff
@@ -1,53 +1,77 @@
...
+        return new OrientDbStudioProperties();
+    }
+    
+    @Bean
+    @ConfigurationProperties(prefix = "strongbox.orientdb.server")
+    OrientDbServerConfiguration orientDbServerProperties() {
```
Comment (carlspring) [inline strongbox-db-import/src/main/java/org/carlspring/strongbox/db/orient/OrientDbConfiguration.java:52]: As above.

File: strongbox-db-import/src/main/resources/application.yml [hunk 7]
```diff
@@ -5,13 +5,19 @@ spring:
...
+      password: password
+      path: @strongbox.orientdb.path@
+    studio:
+      ipAddress: 127.0.0.1
+      port: 2480    
+      enabled: true
+      path: @strongbox.orientdb.path@
 logging:
   level:
     liquibase: INFO
```
Comment (carlspring) [inline strongbox-db-import/src/main/resources/application.yml:23]: Cosmetic: Could you please add a new line here? Thanks! :)

File: strongbox-db-server/src/main/java/org/strongbox/db/server/EmbeddedOrientDbServer.java [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+                                                                                         "http.cache:default",
+                                                                                         "Cache-Control: max-age=120") };
+
+        OServerCommandConfiguration httpCommandConfiguration2 = new OServerCommandConfiguration();
+        httpCommandConfiguration2.implementation = OServerCommandGetGephi.class.getCanonicalName();
+
+        httpListener.commands = new OServerCommandConfiguration[] { httpCommandConfiguration1,
+                                                                    httpCommandConfiguration2 };
+        httpListener.parameters = new OServerParameterConfiguration[] { new OServerParameterConfiguration("utf-8",
+                "network.http.charset") };
```
Comment (carlspring) [inline strongbox-db-server/src/main/java/org/strongbox/db/server/EmbeddedOrientDbServer.java:122]: Cosmetic: Please, re-indent!

File: strongbox-db-server/src/main/java/org/strongbox/db/server/EmbeddedOrientDbServer.java [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+        httpProtocol.name = "http";
+        httpProtocol.implementation = "com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpDb";
+
+        serverConfiguration.network.protocols.add(httpProtocol);
+
+        Path studioPath = Paths.get(studioProperties.getPath()).resolve("studio");
+        if (Files.exists(studioPath))
+        {
+            logger.info(String.format("OrientDB Studio is already available at [%s], skipping initialization. %n" +
+                    "If you want to force the initialization of OrientDB Studio, please remove it's " +
```
Comment (carlspring) [inline strongbox-db-server/src/main/java/org/strongbox/db/server/EmbeddedOrientDbServer.java:136]: Cosmetic: Please, re-indent the parameters!

**Pull request merged**